geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/ClientHealthMonitorIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+import org.apache.geode.cache.server.CacheServer;
+import org.apache.geode.distributed.DistributedSystem;
+import org.apache.geode.internal.AvailablePort;
+import org.apache.geode.internal.cache.EventID;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.test.junit.categories.ClientServerTest;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+
+/**
+ * This is a functional-test for <code>ClientHealthMonitor</code>.
+ */
+@Category({IntegrationTest.class, ClientServerTest.class})
+public class ClientHealthMonitorIntegrationTest {
+  public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+  /**
+   * Default to 0; override in sub tests to add thread pool
+   */
+  protected int getMaxThreads() {
+    return 0;
+  }
+
+  /**
+   * connection proxy object for the client
+   */
+  PoolImpl proxy = null;
+
+  /**
+   * the distributed system instance for the test
+   */
+  DistributedSystem system;
+
+  /**
+   * the cache instance for the test
+   */
+  Cache cache;
+
+  /**
+   * name of the region created
+   */
+  final String regionName = "region1";
+
+  private static int PORT;
+
+  /**
+   * Close the cache and disconnects from the distributed system
+   */
+  @After
+  public void tearDown() throws Exception {
+    if (this.cache != null) {
+      this.cache.close();
+    }
+    if (this.system != null) {
+      this.system.disconnect();
+    }
+    ClientHealthMonitor.shutdownInstance();
+  }
+
+  /**
+   * Initializes proxy object and creates region for client
+   */
+  private void createProxyAndRegionForClient() {
+    try {
+      PoolFactory pf = PoolManager.createFactory();
+      proxy = (PoolImpl) pf.addServer("localhost", PORT).setThreadLocalConnections(true)
+          .setReadTimeout(10000).setPingInterval(10000).setMinConnections(0).create("junitPool");
+      AttributesFactory factory = new AttributesFactory();
+      factory.setScope(Scope.DISTRIBUTED_ACK);
+      cache.createVMRegion(regionName, factory.createRegionAttributes());
+    } catch (Exception ex) {
+      ex.printStackTrace();
+      fail("Failed to initialize client");
+    }
+  }
+
+  private static final int TIME_BETWEEN_PINGS = 2500;
+
+  /**
+   * Creates and starts the server instance
+   */
+  private int createServer() {
+    CacheServer server = null;
+    try {
+      Properties p = new Properties();
+      // make it a loner
+      p.put(MCAST_PORT, "0");
+      p.put(LOCATORS, "");
+
+      this.system = DistributedSystem.connect(p);
+      this.cache = CacheFactory.create(system);
+      server = this.cache.addCacheServer();
+      int port = AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
+      server.setMaximumTimeBetweenPings(TIME_BETWEEN_PINGS);
+      server.setMaxThreads(getMaxThreads());
+      server.setPort(port);
+      server.start();
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail("Failed to create server");
```

Comment:
I find it easier to diagnose problems if the original exception causes the test to fail.  Just have createServer throw Exception and remove the try/catch.

The same applies to createProxyAndRegionForClient.

Reply:
Removed the try/catch